--otp fails error

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

```
❯ npm owner add gar --otp 123456
npm ERR! Cannot read properties of undefined (reading 'error')

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lukekarrys/.npm/_logs/2022-03-28T20_49_30_968Z-debug-0.log
```

### Expected Behavior

It does not fail

### Steps To Reproduce

_No response_

### Environment

_No response_

--- a/lib/utils/otplease.js
+++ b/lib/utils/otplease.js
@@ -1,17 +1,17 @@
-const readUserInfo = require('./read-user-info.js')
-
-module.exports = otplease
 async function otplease (opts, fn) {
   try {
-    await fn(opts)
+    return await fn(opts)
   } catch (err) {
+    const readUserInfo = require('./read-user-info.js')
     if (err.code !== 'EOTP' && (err.code !== 'E401' || !/one-time pass/.test(err.body))) {
       throw err
     } else if (!process.stdin.isTTY || !process.stdout.isTTY) {
       throw err
     } else {
       const otp = await readUserInfo.otp('This operation requires a one-time password.\nEnter OTP:')
-      return fn({ ...opts, otp })
+      return await fn({ ...opts, otp })
     }
   }
 }
+
+module.exports = otplease

--- a/test/lib/utils/otplease.js
+++ b/test/lib/utils/otplease.js
@@ -1,4 +1,5 @@
 const t = require('tap')
+const mockGlobals = require('../../fixtures/mock-globals')
 
 const readUserInfo = {
   otp: async () => '1234',
@@ -6,6 +7,27 @@ const readUserInfo = {
 
 const otplease = t.mock('../../../lib/utils/otplease.js', {
   '../../../lib/utils/read-user-info.js': readUserInfo,
+})
+
+t.test('returns function results on success', async (t) => {
+  const fn = () => 'test string'
+  const result = await otplease({}, fn)
+  t.equal('test string', result)
+})
+
+t.test('returns function results on otp success', async (t) => {
+  mockGlobals(t, {
+    'process.stdin': { isTTY: true },
+    'process.stdout': { isTTY: true },
+  })
+  const fn = ({ otp }) => {
+    if (otp) {
+      return 'success'
+    }
+    throw Object.assign(new Error('nope'), { code: 'EOTP' })
+  }
+  const result = await otplease({}, fn)
+  t.equal('success', result)
 })
 
 t.test('prompts for otp for EOTP', async (t) => {

